blaze/compute/sql.py

```diff
@@ -842,6 +842,24 @@ def compute_up(t, s, **kwargs):
     return s.order_by(*cols)
 
 
+@dispatch(Sample, sa.Table)
+def compute_up(t, s, **kwargs):
+    n = t.n if t.n is not None else int(t.frac * s.count().scalar())
```

Comment:
I don't think sqlalchemy is going to decide any of that. The way to check this would be to output the sql query (you can use `blaze.utils.literal_compile` for this to get any bind params formatted in) and then feed it to `explain` in your db. This will show the query plan which should include any index use